libretro-common/include/libretro.h

```diff
@@ -976,6 +976,19 @@ struct retro_hw_render_context_negotiation_interface
                                             * so it will be used after SET_HW_RENDER, but before the context_reset callback.
                                             */
 
+struct retro_resource
+{
+   const char *rel_path;
+   const char *base_path;
+};
+
+#define RETRO_ENVIRONMENT_GET_RESOURCE_DIRECTORY 44
+                                           /* struct retro_resource * --
+                                            * Interface to acquire resource paths by a relative filename.
+                                            * 'rel_path' should be set to a filename relative to the system directory.
+                                            * 'base_path' will be set to the base path containing the resource or NULL if unknown.
```

Comment:
If the core is allocating the buffer, why not write out the full path? This would allow the frontend to map generic names like "SYSTEM BIOS" into something like "system_bios_foo.bin".